Panic on CRD with image of type "map[interface {}]interface {}"

<!-- 
  Issues without logs and details are more complicated to fix.
  Please help us by filling the template below!
-->

When using skaffold deploy, in my case without even building an image, on a CRD that has the keyword `image`  but as type `map[interface {}]interface {}` skaffold panics.

I have tried using skaffold deploy on a [prometheus-operator CRD](https://github.com/coreos/prometheus-operator/blob/master/contrib/kube-prometheus/manifests/0prometheus-operator-0prometheusCustomResourceDefinition.yaml#L928).

Skaffold should probably never try to replace an `image:` in a CRD, but only in the CR itself.

If i comment out the image in the CRD it works as expected.

### Expected behavior
- Don't panic

### Actual behavior

```
panic: interface conversion: interface {} is map[interface {}]interface {}, not string

goroutine 1 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.(*imageReplacer).NewValue(0xc42061e280, 0x12afbc0, 0xc4201f3920, 0xc42061e201, 0x409715, 0xc4203f4d40)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl/images.go:70 +0x194
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.recursiveVisit(0x12afbc0, 0xc4201f30e0, 0x159dca0, 0xc42061e280)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl/visitor.go:73 +0x1fd
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.recursiveVisit(0x12afbc0, 0xc4201f30b0, 0x159dca0, 0xc42061e280)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl/visitor.go:69 +0x2c9
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.recursiveVisit(0x12afbc0, 0xc4201f3080, 0x159dca0, 0xc42061e280)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl/visitor.go:69 +0x2c9
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.recursiveVisit(0x12afbc0, 0xc420457a70, 0x159dca0, 0xc42061e280)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl/visitor.go:69 +0x2c9
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.recursiveVisit(0x12afbc0, 0xc420457a40, 0x159dca0, 0xc42061e280)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl/visitor.go:69 +0x2c9
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.recursiveVisit(0x12afbc0, 0xc4200b5f20, 0x159dca0, 0xc42061e280)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl/visitor.go:69 +0x2c9
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.recursiveVisit(0x12afbc0, 0xc4200b5e30, 0x159dca0, 0xc42061e280)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl/visitor.go:69 +0x2c9
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.recursiveVisit(0x12afbc0, 0xc4200b5e00, 0x159dca0, 0xc42061e280)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl/visitor.go:69 +0x2c9
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.recursiveVisit(0x12afbc0, 0xc4200b5b60, 0x159dca0, 0xc42061e280)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl/visitor.go:69 +0x2c9
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.recursiveVisit(0x12afbc0, 0xc4200b5110, 0x159dca0, 0xc42061e280)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl/visitor.go:69 +0x2c9
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.(*ManifestList).Visit(0xc4203f5850, 0x159dca0, 0xc42061e280, 0x0, 0x0, 0xc42061e280, 0x0, 0xc4203f5770)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl/visitor.go:45 +0x1a6
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.(*ManifestList).ReplaceImages(0xc4203f5850, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4c211e, ...)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl/images.go:35 +0xa0
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy.(*KubectlDeployer).Deploy(0xc420640000, 0x15b5020, 0xc42004a600, 0x159a9e0, 0xc4200c0008, 0x0, 0x0, 0x0, 0x0, 0x13, ...)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl.go:84 +0x271
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy.(*withLabels).Deploy(0xc4202c7e30, 0x15b5020, 0xc42004a600, 0x159a9e0, 0xc4200c0008, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/labels.go:64 +0xa3
github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner.withTimings.Deploy(0x159db20, 0xc4203ae080, 0x15a49e0, 0xc4205fa100, 0x15b53e0, 0xc4202c7e30, 0x15b5020, 0xc42004a600, 0x159a9e0, 0xc4200c0008, ...)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/timings.go:85 +0x13e
github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner.(*SkaffoldRunner).buildTestDeploy(0xc4201f81b0, 0x15b5020, 0xc42004a600, 0x159a9e0, 0xc4200c0008, 0x0, 0x0, 0x0, 0x15a79e0, 0xc4204780c0)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/runner.go:224 +0x20a
github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner.(*SkaffoldRunner).Run(0xc4201f81b0, 0x15b5020, 0xc42004a600, 0x159a9e0, 0xc4200c0008, 0x0, 0x0, 0x0, 0x1156c62, 0xc4204d7c70)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/runner.go:233 +0x81
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.run(0x159a9e0, 0xc4200c0008, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/run.go:54 +0x152
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.NewCmdDeploy.func1(0xc420476a00, 0x1ebf9f0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/deploy.go:33 +0x33
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).execute(0xc420476a00, 0x1ebf9f0, 0x0, 0x0, 0xc420476a00, 0x1ebf9f0)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:762 +0x468
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1e8bb80, 0xc420476500, 0xc420476280, 0xc420476000)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).Execute(0x1e8bb80, 0xc4200c0008, 0x159a9e0)
	/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run(0x0, 0xc4204d7f78)
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/skaffold.go:27 +0x58
main.main()
	/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/skaffold.go:29 +0x26
```

### Information

- Skaffold version: 

```
skaffold version
v0.19.0
```
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta1
kind: Config
deploy:
  kubectl:
    manifests:
      - 0prometheus-operator-0prometheusCustomResourceDefinition.yaml
```

### Steps to reproduce the behavior

1. `curl -LO https://github.com/coreos/prometheus-operator/raw/master/contrib/kube-prometheus/manifests/0prometheus-operator-0prometheusCustomResourceDefinition.yaml`
2. `skaffold deploy`

--- a/pkg/skaffold/deploy/kubectl/images.go
+++ b/pkg/skaffold/deploy/kubectl/images.go
@@ -67,7 +67,11 @@ func (r *imageReplacer) Matches(key string) bool {
 }
 
 func (r *imageReplacer) NewValue(old interface{}) (bool, interface{}) {
-	image := old.(string)
+	image, ok := old.(string)
+	if !ok {
+		return false, nil
+	}
+
 	found, tag := r.parseAndReplace(image)
 	if !found {
 		subbedImage := r.substituteRepoIntoImage(image)

--- a/pkg/skaffold/deploy/kubectl/images_test.go
+++ b/pkg/skaffold/deploy/kubectl/images_test.go
@@ -126,3 +126,16 @@ func TestReplaceInvalidManifest(t *testing.T) {
 
 	testutil.CheckError(t, true, err)
 }
+
+func TestReplaceNonStringImageField(t *testing.T) {
+	manifests := ManifestList{[]byte(`
+apiVersion: v1
+image:
+- value1
+- value2
+`)}
+
+	output, err := manifests.ReplaceImages(nil, "")
+
+	testutil.CheckErrorAndDeepEqual(t, false, err, manifests.String(), output.String())
+}

